tests/io/s3/test_s3_utils.py

```diff
@@ -54,3 +56,20 @@ def test_s3_open(mocker):
     logger_mock.info.assert_called_once_with("opening mybucket/file.csv")
     assert tmpfile.name.endswith(".s3tmp")
     assert tmpfile.read() == b"a,b\n0,1\n"
+
+
+def test_s3_open_with_token(mocker: MockerFixture) -> None:
+    tempfile_mock = MagicMock()
+    mocker.patch("tempfile.NamedTemporaryFile", return_value=tempfile_mock)
+    mocker.patch("peakina.io.s3.s3_utils._s3_open_file_with_retries")
+    s3fs_file_system = mocker.patch("s3fs.S3FileSystem")
+    s3fs_file_system.return_value.open.return_value = io.BytesIO(b"a,b\n0,1\n")
+    # called with a session_token
+    s3_open("s3://my_key:my_secret@mybucket/file.csv", client_kwargs={"session_token": "xxxx"})
+    s3fs_file_system.assert_called_once_with(
+        token="xxxx", secret="my_secret", key="my_key", client_kwargs={}
```

Comment:
```suggestion
        token="xxxx", secret="my_secret", key="my_key", client_kwargs=None
```

This is waht my rpevious comment was about: If we don't have client kwargs, we should pass `None` here

Reply:
mhhh, What is the issue passing an empty dict there ?

Reply:
https://github.com/ToucanToco/peakina/pull/374/commits/c0a4f8c08fa6f1612baa7be90923fc424e80f59e

Reply:
Two issues:

1. User input is modified, which is a bad practice
2. Here, the assumption is made that for s3fs, `client_kwargs={}` and `client_kwargs=None` has the same behaviour. Nothing guarantees this.

In my opinion, the following logic should be used:

1. If `session_token` is not in `client_kwargs`, just pass the args through as is, without modifying them. If the user passes `{}` rather than `None`, we must assume that they know what they are doing

3. If `session_token` is in `client_kwargs,` create a **new** `client_kwargs` dict with all keys except `session_token`. If that dict is empty, there are no client_kwargs left to pass. In that case, the default value (`None`) should be passed to s3fs.

Reply:
Thanks for the explanations, and you're right !
Can we agree that this commit https://github.com/ToucanToco/peakina/commit/c0a4f8c08fa6f1612baa7be90923fc424e80f59e follow that ?

Reply:
Thanks for the explanations, let me update !

Reply:
No, the commit does not do that: 
* In case `session_token` is **not** in the client kwargs, the code passes `client_kwargs or None`. That's the opposite of the behaviour described in point 1
* In case `session_token` is present, user input is modified, and we might pass an empty dict

Reply:
https://github.com/ToucanToco/peakina/pull/374/commits/0af474c781208fd61e0520b5af9a940780e3fa74

Reply:
> This is waht my rpevious comment was about: If we don't have client kwargs, we should pass None here

this doesn't mean we're changing the user input in some way ?

Reply:
I added more tests just in case : https://github.com/ToucanToco/peakina/pull/374/commits/cb1d442a9774349f903b8b44bf6bce988dd78902